File: app/src/main/java/ru/neosvet/vestnewage/list/CalendarAdapter.java
package ru.neosvet.vestnewage.list;

import android.graphics.Typeface;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import ru.neosvet.vestnewage.R;

public class CalendarAdapter extends RecyclerView.Adapter<CalendarAdapter.ViewHolder> {
    private final List<CalendarItem> data = new ArrayList<>();
    private final View.OnTouchListener click;

    public CalendarAdapter(View.OnTouchListener click) {
        this.click = click;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_calendar, parent, false);
        return new ViewHolder(view);
    }

    public CalendarItem getItem(int i) {
        return data.get(i);
    }

    @Override
    public void onBindViewHolder(CalendarAdapter.ViewHolder holder, int pos) {
        holder.num.setText(data.get(pos).getDay());
        holder.num.setTextColor(data.get(pos).getColor());
        holder.bg.setBackground(data.get(pos).getBG());
        if (data.get(pos).isBold()) {
            holder.num.setTypeface(null, Typeface.BOLD);
        } else {
            holder.num.setTypeface(null, Typeface.NORMAL);
        }
        holder.bg.setTag(pos);
        holder.bg.setOnTouchListener(click);
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public void clear() {
        data.clear();
    }

    public void addItem(CalendarItem calendarItem) {
        data.add(calendarItem);
    }

    public int indexOf(int d) {
        boolean begin = false;
        for (int i = 0; i < data.size(); i++) {
            if (data.get(i).getNum() == 1) {
                if (begin) return -1;
                begin = true;
            }
            if (begin && data.get(i).getNum() == d) {
                return i;
            }
        }
        return -1;
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        View bg;
        TextView num;

        ViewHolder(View itemView) {
            super(itemView);
            bg = itemView.findViewById(R.id.cell_bg);
            num = itemView.findViewById(R.id.cell_num);
        }
    }
}

File: app/src/main/java/ru/neosvet/vestnewage/view/list/CalendarAdapter.kt
package ru.neosvet.vestnewage.view.list

import android.annotation.SuppressLint
import android.graphics.Typeface
import android.view.LayoutInflater
import android.view.MotionEvent
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import ru.neosvet.vestnewage.R
import ru.neosvet.vestnewage.data.CalendarItem

class CalendarAdapter(
    private val clicker: (View, CalendarItem) -> Unit
) : RecyclerView.Adapter<CalendarAdapter.ViewHolder>() {
    private var data = listOf<CalendarItem>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_calendar, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, pos: Int) {
        holder.setItem(data[pos])
    }

    override fun getItemCount(): Int = data.size

    @SuppressLint("NotifyDataSetChanged")
    fun setItems(items: List<CalendarItem>) {
        data = items
        notifyDataSetChanged()
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val num: TextView = itemView.findViewById(R.id.cell_num)

        @SuppressLint("ClickableViewAccessibility")
        fun setItem(item: CalendarItem) {
            num.text = item.text
            num.setTextColor(item.color)
            itemView.background = item.background
            if (item.isBold)
                num.setTypeface(null, Typeface.BOLD)
            else num.setTypeface(null, Typeface.NORMAL)
            itemView.setOnTouchListener { view: View, event: MotionEvent ->
                if (event.action == MotionEvent.ACTION_UP) clicker.invoke(view, item)
                false
            }
        }
    }
}